docroot/modules/humsci/hs_capx/src/Capx.php

```diff
@@ -0,0 +1,230 @@
+
+  /**
+   * CAPx API username.
+   *
+   * @var string
+   */
+  protected $username;
+
+  /**
+   * CAPx API password.
+   *
+   * @var string
+   */
+  protected $password;
+
+  /**
+   * Capx constructor.
+   *
+   * @param \GuzzleHttp\ClientInterface $guzzle
+   *   Guzzle Client service.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cache
+   *   Cache service.
+   * @param \Drupal\Core\Database\Connection $database
+   *   Database connection service.
+   */
+  public function __construct(ClientInterface $guzzle, CacheBackendInterface $cache, Connection $database) {
+    $this->client = $guzzle;
+    $this->cache = $cache;
+    $this->database = $database;
+  }
+
+  /**
+   * Set the CAPx Username.
+   *
+   * @param string $username
+   *   Username.
+   */
+  public function setUsername($username) {
+    $this->username = $username;
+  }
+
+  /**
+   * Set the CAPx Password.
+   *
+   * @param string $password
+   *   Password.
+   */
+  public function setPassword($password) {
+    $this->password = $password;
+  }
+
+  /**
+   * Get the url for CAPx for the given organizations.
+   *
+   * @param string $organizations
+   *   Comma separated organization codes.
+   * @param bool $children
+   *   Include all children of the organizations.
+   *
+   * @return string
+   *   CAPx URL.
+   */
+  public static function getOrganizationUrl($organizations, $children = FALSE) {
+    $organizations = preg_replace('/[^A-Z,]/', '', strtoupper($organizations));
+    $url = self::CAP_URL . "?orgCodes=$organizations&ps=1000";
+    if ($children) {
+      $url .= '&includeChildren=true';
+    }
+    return $url;
+  }
+
+  /**
+   * Get the url for CAPx for given workgroups.
+   *
+   * @param string $workgroups
+   *   Commas separated list of workgroups.
+   *
+   * @return string
+   *   CAPx URL.
+   */
+  public static function getWorkgroupUrl($workgroups) {
+    $workgroups = preg_replace('/[^A-Z,:]/', '', strtoupper($workgroups));
+    return self::CAP_URL . "?privGroups=$workgroups&ps=1000";
+  }
+
+  /**
+   * Test the connection with the username and passwords is valid.
+   *
+   * @return bool
+   *   The connection was successful.
+   */
+  public function testConnection() {
+    $parameters = ['grant_type' => 'client_credentials'];
+
+    try {
+      $response = $this->client->request('GET', self::AUTH_URL, [
+        'query' => $parameters,
+        'auth' => [$this->username, $this->password],
+      ]);
+    }
+    catch (GuzzleException $e) {
+      // Most errors originate from the API itself.
+      return FALSE;
+    }
+
+    // Just in case the API doesn't respond with an error, we also check the
+    // status code.
+    return $response->getStatusCode() == 200;
+  }
+
+  /**
+   * Sync the organization database with the api data from CAP.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   */
+  public function syncOrganizations() {
+    $this->insertOrgData($this->getOrgData());
```

Comment:
Any reason to return a status here?

Reply:
I couldn't find a use case for returning a result. maybe in the future though? I could set it to return an array of all the organization data.